cloudbuild.yaml

```diff
@@ -279,67 +268,72 @@ steps:
 
-#
-# Run the e2e tests without FeatureGates
-#
-
-- name: 'e2e-runner'
-  args:
-    - ''
-    - 'generic'
-    - 'e2e-test-cluster'
-    - 'us-west1-c'
-    - "${_REGISTRY}"
-  id: e2e-stable
-  waitFor:
-    - e2e-feature-gates
-
-#
-# Run the e2e tests on GKE Autopilot with FeatureGates inverted compared to Stable
-#
-# Keep in sync with (the inverse of) pkg/util/runtime/features.go:featureDefaults
-
-- name: 'e2e-runner'
-  args:
-    - 'CustomFasSyncInterval=false&SDKGracefulTermination=false&StateAllocationFilter=false&PlayerAllocationFilter=true&PlayerTracking=true&ResetMetricsOnDelete=true&SafeToEvict=true&PodHostname=true&SplitControllerAndExtensions=true&Example=true'
-    - 'gke-autopilot'
-    - 'gke-autopilot-e2e-test-cluster-1-24'
-    - 'us-west1'
-    - "${_REGISTRY}"
-  id: e2e-feature-gates-gke-autopilot
-  waitFor:
-    - e2e-wait-to-become-leader
+    gcloud builds list --ongoing --filter "tags:'e2e-test'" --format="value(id)" | xargs --no-run-if-empty gcloud builds cancel 
 
-#
-# Run the e2e tests on GKE Autopilot without FeatureGates, except SafeToEvict and SplitControllerAndExtensions, which we need on Autopilot
-#
+# kick off the child e2e test cloud builds in parallel, fail this build if any of the child build fails
 
-- name: 'e2e-runner'
-  args:
-    - 'SafeToEvict=true&SplitControllerAndExtensions=true'
-    - 'gke-autopilot'
-    - 'gke-autopilot-e2e-test-cluster-1-24'
-    - 'us-west1'
-    - "${_REGISTRY}"
-  id: e2e-stable-gke-autopilot
+- name: 'gcr.io/cloud-builders/gcloud'
+  id: submit-e2e-test-cloud-build
+  entrypoint: bash
+  args: 
+   - '-c'
+   - |
+      set -e
+      pids=()
+      cloudProducts=("generic" "gke-autopilot")
+      for cloudProduct in ${cloudProducts[@]}
+      do
+        if [ $cloudProduct = generic ]
+        then 
+          featureWithGate="CustomFasSyncInterval=false&SDKGracefulTermination=false&StateAllocationFilter=false&PlayerAllocationFilter=true&PlayerTracking=true&ResetMetricsOnDelete=true&SafeToEvict=true&PodHostname=true&SplitControllerAndExtensions=true&Example=true"
+          featureWithoutGate=""
+          testClusterLocation="us-west1-c"
+          testCluster="e2e-test-cluster"
+        else
+          featureWithGate="CustomFasSyncInterval=false&SDKGracefulTermination=false&StateAllocationFilter=false&PlayerAllocationFilter=true&PlayerTracking=true&ResetMetricsOnDelete=true&SafeToEvict=true&PodHostname=true&SplitControllerAndExtensions=true&Example=true"
+          featureWithoutGate="SafeToEvict=true&SplitControllerAndExtensions=true"
+          testClusterLocation="us-west1"
+          testCluster="gke-autopilot-e2e-test-cluster-1-24"
+        fi
+        # TODO: Add more versions in the list, and generate the test cluster name in the loop rather than hardcode once the clusters are created
+        k8sVersions=("1.24")
+        for k8sVersion in ${k8sVersions[@]}
+        do
+          gcloud builds submit . --config=e2e-test-cloudbuild.yaml \
```

Comment:
Hmm, I'm looking at https://pantheon.corp.google.com/cloud-build/builds;region=global/e36f952d-efc8-488d-b79d-49cee84bf285;step=23?project=agones-images and I see one issue: The output is all spliced together.

Even if I go to the top of that log and flip to https://console.cloud.google.com/cloud-build/builds/5f8581b6-cfc7-4873-9cd5-cf15ee67b30f?project=258182270954, it's not obvious which build is running at that point.

Putting yourself in the shoes of a developer looking trying to understand failed e2e, can we make sure this is less toily?

I wonder, and it's a little bit nuts, if the steps in the child build can have substitutions in them? So for example instead of `e2e-stable` they become `e2e-stable-${CLOUD_PRODUCT}-${K8S_VERSION}` or something - then even this output in the main build would be understandable.

Reply:
After some research and consultation with Cloud Build folks, it turns out it is not possible to use substitutions in the `id` field of build config file. At this point, I feel there is no way to improve the main cloud build output and developer will need to go to the failed child e2e cloud build to see details about failed test. Maybe we can do something to the go test output to include the cluster name? But I'm not sure whether that's feasible either. What's your thought?

Reply:
Here is my thought for something lightweight:

* change line 311 to e.g.:

```
gcloud builds submit [...] |& sed -e "s/^/${cloudProduct}-${k8sVersion}: /"
```

this should prefix every line of the output with e.g. `autopilot-1.24: `, including the line at the top of the output that lists the links off to other builds.

* I would additionally add some output to the front of `entrypoint.sh` to all the incoming args. I think even just something like `echo e2e-runner arguments: "$*"` would be good enough - something so that if you're actually looking at a sub-build, you can decipher what config it is quickly.

Reply:
Oh, I didn't realize the build log in the console is actually the `gcloud builds submit` output. I will try this out.

Reply:
Output looks great after [4924d3b](https://github.com/googleforgames/agones/pull/2949/commits/4924d3bdd356b80c2a951c8d634a161aef6e1c55)!

I'm actually going to rescind the second bullet. This looks clean enough I think we can just rely on it. If you happen to browse to one of the sub-builds, you can view the substitutions in the `Execution Detail` tab as well: https://screenshot.googleplex.com/n8dwbbJkVEJ8Bdh .. and since the links are the console links anyways, I think those links will only really work for us. However, the GCS logs agones-bot lists will have the interleaved logs, so it should be possible to figure out what configuration failed regardless.